sync_members leaks extra.annotations across Member instances via shared class-level dict

## Summary

`Member.overlay()` mutates a class-level mutable default dict, causing `extra.annotations` (and other fields) written to one member to leak into every other Member instance. This surfaces as a hard bug when `sync_members` is enabled: PRs generated by the action copy annotations from one member onto unrelated members.

Reproduced against tags `20260331` and `20260413`. Still present on `main` (`db0792f`).

## Observed impact

Downstream example — [camaraproject/camara-landscape#282](https://github.com/camaraproject/camara-landscape/pull/282) adds this block to 8 members that had no annotations and overwrites a 9th member's existing annotations:

```yaml
extra:
  annotations:
    contact: networkapi@vodafone.com            # member A's value
    demo_url: https://developer.vodafone.com/    # member A's value
    demo_url2: https://www.youtube.com/watch?v=YCj7Vo5RDek  # member B's value
```

The mix of fields from two different members is the signature of a field-level accumulator, not a simple "wrong member copied" bug.

## Root cause

`lfx_landscape_tools/member.py` declares mutable containers at class level:

```python
class Member:
    ...
    second_path = []
    organization = {}
    __extra = {}
    ...
    additional_repos = []
```

`Member.__init__` does not re-bind these, so every fresh `Member()` returns the same underlying dict/list via `getattr`. In `Member.overlay()`:

```python
if isinstance(val, dict):
    base = curr if isinstance(curr, dict) else {}
    self._update_nested_dict(key, base, val)
    setattr(self, key, base)
```

`curr = getattr(self, 'extra')` returns the shared class dict when no instance attribute exists yet. `_update_nested_dict` mutates `base` (= the class dict) in place via `sub_dict[s_skey] = s_sval`. On the next Member, `curr` again returns that same now-polluted class dict, so each overlay accumulates onto the previous. At serialization, `toLandscapeItemAttributes()` reads `self.extra` and sees the accumulated values on every instance.

The `extra` setter assigns `self.__extra = endextra` to a fresh dict, which creates an instance attribute — but only AFTER `_update_nested_dict` has already mutated the aliased class dict, so the damage is done.

The same pattern also affects `second_path`, `organization`, and `additional_repos`.

## Minimal reproducer

```python
from lfx_landscape_tools.member import Member

a = Member(); a.name = 'a'; a.homepage_url = 'https://a.example/'
b = Member(); b.name = 'b'; b.homepage_url = 'https://b.example/'

src = Member()
src.name = 'a'
src.extra = {'annotations': {'contact': 'a@example.com'}}
a.overlay(src)

assert b.extra == {}, f"leak: {b.extra}"
# AssertionError: leak: {'annotations': {'contact': 'a@example.com'}}
```

## Proposed fix

Move the mutable defaults into `__init__` so each instance has its own container. PR forthcoming.

## Related

- Sync feature introduced in #307, tagged `20260331`.
- The restructure of flat `extra.contact`/`extra.demo_url` into `extra.annotations.*` (via the `extra` setter) is unrelated to the leak but explains why the leaked fields materialize under `annotations`.

--- a/lfx_landscape_tools/member.py
+++ b/lfx_landscape_tools/member.py
@@ -30,12 +30,8 @@ from lfx_landscape_tools.svglogo import SVGLogo
 class Member:
 
     membership = None
-    second_path = []
-    organization = {}
-    __extra = {}
     project = None
     project_org = None
-    additional_repos = []
     __name = None
     __homepage_url = None
     __logo = None
@@ -51,6 +47,15 @@ class Member:
     itemschema = []
 
     def __init__(self):
+        # Per-instance containers for mutable fields. Declaring these at
+        # class level would alias a single dict/list across every Member
+        # instance, causing values written to one member to surface on
+        # others (see sync_members leak symptoms).
+        self.second_path = []
+        self.organization = {}
+        self.__extra = {}
+        self.additional_repos = []
+
         # load in data schema from landscape2
         try:
             schemaURL = 'https://raw.githubusercontent.com/cncf/landscape2/refs/heads/main/docs/config/data.yml'

--- a/test/test_member.py
+++ b/test/test_member.py
@@ -714,6 +714,42 @@ class TestMember(unittest.TestCase):
         with unittest.mock.patch("lfx_landscape_tools.svglogo.open", unittest.mock.mock_open(read_data="data")) as mock_file:
             membertooverlay.overlay(member)
 
+    def testExtraIsolationBetweenInstances(self):
+        # Regression: writing extras via overlay on one Member must not
+        # leak to other Member instances through a shared class-level dict.
+        a = Member()
+        a.name = 'a'
+        a.homepage_url = 'https://a.example/'
+
+        b = Member()
+        b.name = 'b'
+        b.homepage_url = 'https://b.example/'
+
+        source = Member()
+        source.name = 'a'
+        source.extra = {'annotations': {'contact': 'a@example.com'}}
+        a.overlay(source)
+
+        self.assertEqual(a.extra.get('annotations', {}).get('contact'), 'a@example.com')
+        self.assertEqual(b.extra, {})
+
+        c = Member()
+        c.name = 'c'
+        c.homepage_url = 'https://c.example/'
+        source2 = Member()
+        source2.name = 'c'
+        source2.extra = {'annotations': {'demo_url2': 'https://example.com/demo'}}
+        c.overlay(source2)
+
+        self.assertNotIn('contact', c.extra.get('annotations', {}))
+        self.assertEqual(c.extra.get('annotations', {}).get('demo_url2'), 'https://example.com/demo')
+
+        d = Member()
+        self.assertEqual(d.extra, {})
+        self.assertEqual(d.second_path, [])
+        self.assertEqual(d.organization, {})
+        self.assertEqual(d.additional_repos, [])
+
     @responses.activate
     def testHostLogo(self):
         with tempfile.TemporaryDirectory() as tempdir:

